src/data/CCBV.ml

```diff
@@ -118,34 +155,34 @@ let get bv i =
 *)
 
 let set bv i =
-  let n = i / __width in
-  if n >= Array.length bv.a
-  then resize bv (n+1);
-  let i = i - n * __width in
-  bv.a.(n) <- bv.a.(n) lor (1 lsl i)
+  if i < 0 then invalid_arg "set: negative index" else
```

Comment:
see the "dangling else" here? for an exception-throwing `if` I'd do:

```ocaml
if i < 0 then invalid_arg "foo";
…
```

but apart from that, this else should really be wrapped into `( … )`, because of the imperative `;` below.